spec/models/mhv_account_spec.rb

```diff
@@ -0,0 +1,174 @@
+          subject = described_class.new(base_attributes)
+          subject.send(:setup) # This gets called when object is first loaded
+          expect(subject.account_state).to eq('unknown')
+        end
+      end
+
+      context 'with terms not accepted' do
+        it 'is ineligible if not a va patient' do
+          allow_any_instance_of(User).to receive(:icn).and_return(nil)
+          subject = described_class.new(user_uuid: user.uuid, account_state: 'needs_terms_acceptance')
+          subject.send(:setup) # This gets called when object is first loaded
+          expect(subject.account_state).to eq('ineligible')
+        end
+
+        it 'transitions to needs_terms_acceptance' do
+          subject = described_class.new(user_uuid: user.uuid, account_state: 'upgraded', upgraded_at: Time.current)
+          subject.send(:setup) # This gets called when object is first loaded
+          expect(subject.account_state).to eq('needs_terms_acceptance')
+        end
+
+        it 'is able to transition back to registered' do
+          subject = described_class.new(user_uuid: user.uuid, account_state: 'registered', registered_at: Time.current)
+          subject.send(:setup) # This gets called when object is first loaded
+          expect(subject.account_state).to eq('needs_terms_acceptance')
+        end
+
+        it 'it falls back to unknown' do
+          subject = described_class.new(user_uuid: user.uuid, account_state: 'unknown')
+          subject.send(:setup) # This gets called when object is first loaded
+          expect(subject.account_state).to eq('needs_terms_acceptance')
+        end
+      end
+    end
+  end
+
+  describe 'account creation and upgrade' do
+    before(:each) do
+      terms = create(:terms_and_conditions, latest: true, name: described_class::TERMS_AND_CONDITIONS_NAME)
+      create(:terms_and_conditions_acceptance, terms_and_conditions: terms, user_uuid: user.uuid)
+    end
+
+    subject { described_class.new(user_uuid: user.uuid) }
+
+    it 'will create and upgrade an account' do
+      expect(subject.terms_and_conditions_accepted?).to be_truthy
+      expect(subject.preexisting_account?).to be_falsey
+      expect(subject.persisted?).to be_falsey
+      expect(user.mhv_correlation_id).to be_nil
+      VCR.use_cassette('mhv_account_creation/creates_an_account') do
+        VCR.use_cassette('mhv_account_creation/upgrades_an_account') do
+          subject.create_and_upgrade!
+          expect(subject.persisted?).to be_truthy
+          expect(subject.account_state).to eq('upgraded')
+          expect(subject.registered_at).to be_a(Time)
+          expect(subject.upgraded_at).to be_a(Time)
+          expect(User.find(user.uuid).mhv_correlation_id).to eq('14221465')
+        end
+      end
+    end
+
+    context 'existing account that has not been upgraded' do
+      let(:mhv_ids) { ['14221465'] }
+
+      it 'will only upgrade an account if no mhv_correlation_id exists' do
```

Comment:
is this spec description accurate? the "if not mhv_correlation_id exists" part doesn't make sense for upgrade case, right?